src/txHashBuilder.c

```diff
@@ -7,6 +7,18 @@
 #include "crc32.h"
 #include "bufView.h"
 
+// this tracing is rarely needed
+// so we want to keep it turned off to avoid polluting the trace log
+
+//#define TRACE_TX_HASH_BUILDER
+
+#ifdef TRACE_TX_HASH_BUILDER
```

Comment:
what's the benefit of such tracing? Even if it was there I wouldn't really be able to tell at first sight what's it for so I'd most likely skip it even if was actually beneficial

Reply:
You probably never tried to debug tx hash builder :) It has its own state machine, so most changes in the sign tx state machine are reflected for tx hash builder, too. If you mess it up, as I did this time, tracing is very handy, and without this, you would have to write it ad-hoc (and waste some time by repeatedly deploying code to ledger which gets stuck every time, just to get the tracing right).

On the other hand, I don't want to have the tracing active all the time, because it pollutes the log (data are added to the tx hash builder often in small bits). Usually the tx hash builder is easy to fix, time devoted to that is, say, 5-10% of the total time spent on modifying signTx.